pkg/querier/queryrange/roundtrip.go

```diff
@@ -64,7 +64,7 @@ func NewTripperware(
 		return nil, nil, err
 	}
 
-	seriesTripperware, err := NewSeriesTripperware(cfg, log, limits, lokiCodec, instrumentMetrics, retryMetrics, splitByMetrics)
+	seriesTripperware, cache, err := NewSeriesTripperware(cfg, log, limits, lokiCodec, SeriesExtractor{}, instrumentMetrics, retryMetrics, splitByMetrics)
```

Comment:
This is overwriting the previous metrics cache. You'll need to make a `type MultiStopper []Stopper` type or similar in order to coalesce the multiple caches that are returned.

Reply:
Sure. Will make the change